packages/mookme/src/utils/hook-package.ts

```diff
@@ -21,29 +40,78 @@ export function hookPackage(hook: PackageHook): Promise<{ hook: PackageHook; ste
     venvActivate: hook.venvActivate,
   };
 
-  return new Promise((resolve, reject) =>
-    Promise.all(hook.steps.map((step) => runStep(step, options)))
-      .then((errors) => {
-        const hasError = errors.find((err) => err !== null);
-        const result = hasError ? chalk.bgRed.bold(' Error × ') : chalk.bgGreen.bold(' Done ✓ ');
-        loggers[hook.name](`${chalk.bold.inverse(` Hooks : ${hook.name} `)}${result}`);
-        resolve(
-          errors
-            .filter((err) => err !== null)
-            .map((err) => ({
+  const promises = [];
+  const errors: { hook: PackageHook; step: StepCommand; msg: Error }[] = [];
+
+  for (const step of hook.steps) {
+    try {
+      const loaderManager = ui.stepsLoggers[step.name];
+      const stepPromise = runStep(step, options, loaderManager);
+      promises.push(stepPromise);
+
+      if (step.serial) {
+        const stepError = await stepPromise;
+        let result: string = chalk.bgGreen.bold(' Done ✓ ');
+        if (stepError !== null) {
+          result = chalk.bgRed.bold(' Error × ');
+          errors.push({
+            hook,
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            step: stepError!.step,
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            msg: stepError!.msg,
+          });
+        }
+        ui.packageLogger(`${chalk.bold.inverse(` Hooks : ${hook.name} `)}${result}`);
+      } else {
+        stepPromise.then((stepError) => {
+          let result: string = chalk.bgGreen.bold(' Done ✓ ');
+          if (stepError !== null) {
+            result = chalk.bgRed.bold(' Error × ');
+            errors.push({
               hook,
               // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-              step: err!.step,
+              step: stepError!.step,
               // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-              msg: err!.msg,
-            })),
-        );
-      })
-      .catch((err) => {
-        loggers[hook.name](`${chalk.bold.inverse(` Hooks : ${hook.name} `)}${chalk.bgRed.bold(' Error × ')}`);
-        reject(err);
-      }),
-  );
+              msg: stepError!.msg,
+            });
+          }
+          ui.packageLogger(`${chalk.bold.inverse(` Hooks : ${hook.name} `)}${result}`);
+        });
+      }
+    } catch (err) {
+      ui.packageLogger(`${chalk.bold.inverse(` Hooks : ${hook.name} `)}${chalk.bgRed.bold(' Error × ')}`);
+      throw err;
+    }
+  }
+
+  await Promise.all(promises);
+
+  return errors;
+
+  // return await new Promise((resolve, reject) =>
```

Comment:
Any reason to include this?

Reply:
Nope ! this is dead code I forgot to remove